app/services/post_scraper.rb

```diff
@@ -1,88 +1,59 @@
+    @board_id = board_id
     @section_id = section_id
-    @status = status || Post::STATUS_COMPLETE
-    url += (url.include?('?') ? '&' : '?') + 'style=site' unless url.include?('style=site')
-    url += '&view=flat' unless url.include?('view=flat') || threaded_import
-    @url = url
-    @console_import = console_import
-    @threaded_import = threaded_import # boolean
+    @status = status
+    @console_import = console
+    @threaded_import = threaded # boolean
     @subject = subject
+    @url = clean_url(url)
+    super()
   end
 
-  def scrape!
-    @html_doc = doc_from_url(@url)
+  def scrape(threads=nil)
+    html_doc = doc_from_url(@url)
 
     Post.transaction do
-      import_post_from_doc(@html_doc)
-      import_replies_from_doc(@html_doc)
-      links = page_links
-      links.each_with_index do |link, i|
-        logger.debug "Scraping '#{@post.subject}': page #{i+1}/#{links.count}"
-        doc = doc_from_url(link)
-        import_replies_from_doc(doc)
+      import_post_from_doc(html_doc)
+      if threads.present?
+        threads.each { |thread| evaluate_links(doc_from_url(thread)) }
+      else
+        evaluate_links(html_doc)
       end
       finalize_post_data
     end
+    return if @errors.present?
     GenerateFlatPostJob.perform_later(@post.id)
     @post
   end
+  alias scrape! scrape
 
-  # works as an alternative to scrape! when you want to scrape particular top-level threads of a post sequentially
+  # works as an alternative to scrape when you want to scrape particular top-level threads of a post sequentially
   # "threads" are URL permalinks to the threads to scrape, which it will scrape in the given order
-  def scrape_threads!(threads)
-    raise RuntimeError.new('threaded_import must be true to use scrape_threads!') unless @threaded_import
+  def scrape_threads(threads)
+    @errors.add(:base, 'threaded_import must be true to use scrape_threads') && return unless @threaded_import
+    scrape(threads)
+  end
+  alias scrape_threads! scrape_threads
 
-    @html_doc = doc_from_url(@url)
-    # if threads.blank?
-    #   threads = top_level_comments(@html_doc)
-    # end
+  private
 
-    Post.transaction do
-      import_post_from_doc(@html_doc)
-      threads.each do |thread|
-        @html_doc = doc_from_url(thread)
-        import_replies_from_doc(@html_doc)
-        links = page_links
-        links.each_with_index do |link, i|
-          logger.debug "Scraping '#{@post.subject}': page #{i+1}/#{links.count}"
-          doc = doc_from_url(link)
-          import_replies_from_doc(doc)
-        end
-      end
-      finalize_post_data
+  attr_writer :errors
+
+  def evaluate_links(base_doc)
+    import_replies_from_doc(base_doc)
+    links = page_links(base_doc)
+    links.each_with_index do |link, i|
+      Resque.logger.debug "Scraping '#{@post.subject}': page #{i+1}/#{links.count}"
```

Comment:
What's up with all the `logger.debug` (etc) calls turning into `Resque.logger.debug`? Did something change?

Reply:
logger was a method on PostScraper. I can revert that if you want?

Reply:
Ah! It's probably fine.